File: src/main/java/com/lsd/properties/LsdProperties.java
package com.lsd.properties;

import java.util.Properties;

import static com.lsd.properties.DefaultProperties.defaultProperties;
import static java.lang.Boolean.parseBoolean;
import static java.lang.Integer.parseInt;

/**
 * Convenience class to load up all the properties (defaults and user provided) at the start and make available globally
 */
public class LsdProperties {
    public static final String DIAGRAM_THEME = "lsd.core.diagram.theme";
    public static final String LABEL_MAX_WIDTH = "lsd.core.label.maxWidth";
    public static final String OUTPUT_DIR = "lsd.core.report.outputDir";
    public static final String DETERMINISTIC_IDS = "lsd.core.ids.deterministic";

    private static final Properties PROPERTIES = new PropertiesFileLoader("lsd.properties", defaultProperties()).load();

    public static String get(String key) {
        return PROPERTIES.getProperty(key);
    }

    public static int getInt(String key) {
        return parseInt(PROPERTIES.getProperty(key));
    }

    public static boolean getBoolean(String key) {
        return parseBoolean(PROPERTIES.getProperty(key));
    }
}

File: src/main/kotlin/com/lsd/core/properties/LsdProperties.kt
package com.lsd.core.properties

/**
 * Convenience class to load up all the properties (defaults and user provided) at the start and make available globally
 */
object LsdProperties {
    const val DIAGRAM_THEME = "lsd.core.diagram.theme"
    const val LABEL_MAX_WIDTH = "lsd.core.label.maxWidth"
    const val OUTPUT_DIR = "lsd.core.report.outputDir"
    const val DETERMINISTIC_IDS = "lsd.core.ids.deterministic"
    const val MAX_EVENTS_PER_DIAGRAM = "lsd.core.diagram.sequence.maxEventsPerDiagram"
    const val DEV_MODE = "lsd.core.devMode"
    const val METRICS_ENABLED = "lsd.core.metrics.enabled"

    private val PROPERTIES = PropertiesLoader("lsd.properties", DefaultProperties.defaultProperties()).load()

    @JvmStatic
    operator fun get(key: String): String {
        return PROPERTIES.getProperty(key)
    }

    @JvmStatic
    operator fun get(key: String, default: String): String {
        return PROPERTIES.getProperty(key, default)
    }

    @JvmStatic
    fun getInt(key: String): Int {
        return PROPERTIES.getProperty(key).toInt()
    }

    @JvmStatic
    fun getInt(key: String, default: Int): Int {
        return PROPERTIES.getProperty(key, default.toString()).toInt()
    }

    @JvmStatic
    fun getLong(key: String): Long {
        return PROPERTIES.getProperty(key).toLong()
    }

    @JvmStatic
    fun getLong(key: String, default: Long): Long {
        return PROPERTIES.getProperty(key, default.toString()).toLong()
    }

    @JvmStatic
    fun getBoolean(key: String): Boolean {
        return PROPERTIES.getProperty(key).toBoolean()
    }

    @JvmStatic
    fun getBoolean(key: String, default: Boolean): Boolean {
        return PROPERTIES.getProperty(key, default.toString()).toBoolean()
    }
}